compiler/qsc_eval/src/tests.rs

```diff
@@ -2148,6 +2148,121 @@ fn update_range_to_end() {
     check_expr("", "...2..3 w/ End <- 10", &expect!["...2..10"]);
 }
 
+#[test]
+fn update_array_with_range() {
+    check_expr("", "[0, 1, 2] w/ 1..2 <- [10, 11]", &expect!["[0, 10, 11]"]);
+}
+
+#[test]
+fn update_array_with_range_start() {
+    check_expr(
+        "",
+        "[0, 1, 2, 3] w/ 1... <- [10, 11]",
+        &expect!["[0, 10, 11, 3]"],
+    );
+}
+
+#[test]
+fn update_array_with_range_step() {
+    check_expr(
+        "",
+        "[0, 1, 2, 3] w/ ...2... <- [10, 11]",
+        &expect!["[10, 1, 11, 3]"],
+    );
+}
+
+#[test]
+fn update_array_with_range_end() {
+    check_expr(
+        "",
+        "[0, 1, 2, 3] w/ ...2 <- [10, 11, 12, 13]",
```

Comment:
Oh interesting, the size of the new value array can be too big for the range. Can you add some tests for the following scenarios:
- new values array is too small for open-ended range
- new values array is too big for close-ended range
- new values array is too small for close-ended range
- new values array is empty

Reply:
Sure thing! I think some of these cases are covered here and there in these tests, but not exhaustively. I'll use your list here to double check and make sure I have individual tests for all of them.